filebeat/crawler/prospector_log.go

```diff
@@ -48,32 +46,37 @@ func (p *ProspectorLog) Init() {
 
 		// Offset found -> skip to previous state
 		if offset > 0 {
-			fs := input.NewFileStat(fileinfo, 0)
+			fs := input.NewFileState(fileinfo)
 			fs.Skip(offset)
-			p.harvesterStats[path] = *fs
+			p.harvesterStates[path] = *fs
```

Comment:
On line 149 it is still passed as reference. Not 100% sure if that is necessary, but will clean this up too. One thing I just realised is that the naming of variables it quite confusing now (`newInfo` vs `newState`). Will change that in this PR.